vignettes/extending-jmpost.Rmd

```diff
@@ -28,6 +28,151 @@ library(jmpost)
 Please note that this document is currently a work-in-progress and does not
 contain complete information for this package yet. 
 
+
+## Custom Link Functions
+
+Users can defined custom link functions in several ways based upon the level of customisation
+required. In order to explain this process it is first important to understand how the link
+functions are implemented under the hood.
+
+The link functions add their contribution to the likelihood function via the log-hazard function;
+that is the general model is formulated as:
+
+$$
+log(h_i(t, \phi_i)) = log(h_0(t)) + X_i \beta + \alpha_1 f(t, \phi_i) + \alpha_2 g(t, \phi_i) + \dots
+$$
+
+Where:
+- $X$ is the design matrix of covariates
+- $\beta$ is the vector of coefficients for the covariates
+- $f(t)$ and $g(t)$ are the link functions
+- $\alpha_1$ and $\alpha_2$ are the coefficients for the link functions
+- $h_0(t)$ is the baseline hazard function
+- $\phi_i$ is a vector of parameters from the longitudinal model
+
+Each longitudinal model is responsible for defining their own implementations of the $\phi$ vector.
+The interface for doing this is by providing an `enableLink` method that updates the `StanModule`
+object of the model to define a Stan matrix with the name `link_function_inputs`
+that contains 1 row per
+patient and 1 column per $\phi$ parameter.
+
+For reference the following is roughly the implementation for the `LongitudinalGSF` model:
+
+```R
+enableLink.LongitudinalGSF <- function(object, ...) {
+  stan <- StanModule("
+transformed parameters {
+    matrix[Nind, 4] link_function_inputs;
+    link_function_inputs[,1] = lm_gsf_psi_bsld;
+    link_function_inputs[,2] = lm_gsf_psi_ks;
+    link_function_inputs[,3] = lm_gsf_psi_kg;
+    link_function_inputs[,4] = lm_gsf_psi_phi;
+}")
+    object@stan <- merge(
+        object@stan,
+        stan
+    )
+    object
```

Comment:
side comment (not for this PR): it might be nice to have an `update(object, stan)` method that does this, in a bit more readable and concise way